lib/Varien/Db/Adapter/Pdo/Mysql.php

```diff
@@ -1643,9 +1637,7 @@ public function resetDdlCache($tableName = null, $schemaName = null)
         }
         if ($tableName === null) {
             $this->_ddlCache = [];
-            if ($this->_cacheAdapter instanceof Zend_Cache_Core) {
-                $this->_cacheAdapter->clean(Zend_Cache::CLEANING_MODE_MATCHING_TAG, [self::DDL_CACHE_TAG]);
-            }
+            $this->_cacheAdapter->clean([self::DDL_CACHE_TAG]);
```

Comment:
```suggestion
            if ($this->_cacheAdapter) {
                $this->_cacheAdapter->clean([self::DDL_CACHE_TAG]);
            }
```

I don't know under which circumstance we wouldn't have `$_cacheAdapter`, but we left the checks everywhere else  so I think we should add here.